controllers/sink.go

```diff
@@ -58,11 +58,13 @@ func (r *SinkReconciler) ObserveSinkStatefulSet(ctx context.Context, req ctrl.Re
 	// statefulset created, waiting it to be ready
 	condition.Action = v1alpha1.Wait
 
-	if *statefulSet.Spec.Replicas != *sink.Spec.Replicas {
+	if *statefulSet.Spec.Replicas != *sink.Spec.Replicas ||
+		statefulSet.Spec.ServiceName != spec.MakeSinkServiceName(sink) {
```

Comment:
The `serviceName` of the `StatefulSet` may not be able to be updated at the moment, otherwise you may get a error message like `spec: Forbidden: updates to statefulset spec for fields other than 'replicas', 'template', and 'updateStrategy' are forbidden`. The only possible way to update it might be delete the `StatefulSet` then recreate it. It might be a non-trivial change and we can leave it future PRs.

Reply:
https://github.com/streamnative/function-mesh/issues/222 to track this issue